modules/apps/object/object-service/src/main/java/com/liferay/object/internal/action/util/ObjectEntryVariablesUtil.java

```diff
@@ -115,6 +126,55 @@ public static Map<String, Object> getActionVariables(
 		return variables;
 	}
 
+	public static Map<String, Object> getDDMValidationRuleVariables(
+			BaseModel<?> baseModel, DTOConverterRegistry dtoConverterRegistry,
+			ObjectDefinition objectDefinition,
+			ObjectEntryLocalService objectEntryLocalService,
+			JSONObject payloadJSONObject,
+			SystemObjectDefinitionMetadataRegistry
+				systemObjectDefinitionMetadataRegistry)
+		throws PortalException {
+
+		if (PropsValues.OBJECT_ENTRY_SCRIPT_VARIABLES_VERSION == 2) {
+			return HashMapBuilder.<String, Object>put(
+				"objectEntry",
+				_getVariables(
+					dtoConverterRegistry, objectDefinition, payloadJSONObject,
+					systemObjectDefinitionMetadataRegistry, false)
+			).put(
+				"originalObjectEntry",
+				_getVariables(
+					dtoConverterRegistry, objectDefinition, payloadJSONObject,
+					systemObjectDefinitionMetadataRegistry, true)
+			).build();
+		}
+
+		Map<String, Object> variables = HashMapBuilder.<String, Object>putAll(
+			baseModel.getModelAttributes()
+		).build();
+
+		if (baseModel instanceof ObjectEntry) {
+			variables.putAll(
+				objectEntryLocalService.getValues((ObjectEntry)baseModel));
+		}
+		else {
+			Map<String, Serializable> extendedProperties =
+				EntityExtensionThreadLocal.getExtendedProperties();
+
+			if (extendedProperties != null) {
+				variables.putAll(extendedProperties);
+			}
+		}
+
+		variables.putAll(
+			objectEntryLocalService.
+				getExtensionDynamicObjectDefinitionTableValues(
+					objectDefinition,
+					GetterUtil.getLong(baseModel.getPrimaryKeyObj())));
+
+		return variables;
```

Comment:
@p-albuquerque this code is duplicated in `getValidationRuleVariables`. Can you create a private method to reuse this code?